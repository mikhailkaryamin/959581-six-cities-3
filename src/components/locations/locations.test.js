import React from "react";
import renderer from "react-test-renderer";
import Locatons from "./locations.jsx";
import configureStore from "redux-mock-store";
import {
  Provider
} from "react-redux";

const mockStore = configureStore([]);

const handleLocationClick = () => {};
it(`Render locations correctly`, () => {
  const store = mockStore({
    locations: [
      `Paris`,
      `Cologne`,
      `Brussels`,
      `Amsterdam`,
      `Hamburg`,
      `Dusseldorf`
    ]
  });
  const tree = renderer
    .create(
        <Provider store={
          store
        }>
          <Locatons
            handleLocationClick={
              handleLocationClick
            }
          />
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
