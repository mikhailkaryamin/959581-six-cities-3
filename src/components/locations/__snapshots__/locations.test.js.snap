// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render locations correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Cologne
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Hamburg
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </li>
  </ul>
</section>
`;
